### ISSUE TITLE
unable to add token

### ISSUE THREAD

**Issue by Startupeers** [MetaMask/metamask-extension#2675]
I can't add any token to my metamask account / plugin.

For example, when I try to add Golem (GNT 0xa74476443119A942dE498590Fe1f2454d7D4aC0d) the token contract address is recognized because Token symbol is replaced automaticaly with GNT but when I click on "Add" nothing happen.
It stil show "No tokens found" :(

**Comment by fraser1808** [comment]
Im having this same problem. Ive tried with multiple contact addresses but each time i click add, i can see the loading symbol for half a second but then nothing happens. PLEASE help.

**Comment by Christophe80** [comment]
Same problem here... I see my tokens in etherscan.io but I am unable to add them... If I do, everything fills in correct, but when I click add, I get a message "token not found" ... does domeone has a solution pls?

**Comment by JeffWScott** [comment]
Same issue

**Comment by bgadrian** [comment]
FF 59.a1 on Win10
![video](https://i.imgur.com/Pgz5r1o.gif)

I couldn't find any error in the console (javascript) when I press Add and no requests in the Network panel.

`Could not map contract ID '@mozilla.org/fxaccounts/push;1' to CID {1b7db999-2ecd-4abf-bb95-a726896798ca} because no implementation of the CID is registered.  components.manifest:176
Successfully compiled asm.js code (total compilation time 1ms; not stored in cache (too small to benefit))  adblockplus.js
XML Parsing Error: not well-formed
Location: moz-extension://aa445b6e-602f-4e59-a87f-0a4dd5681e66/content_scripts/vimium.css
Line Number 1, Column 1:  vimium.css:1:1
Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help http://xhr.spec.whatwg.org/  background.js:1:434906
unreachable code after return statement[Learn More]  Bookmarks.js:1639:8
Content Security Policy: The page's settings blocked the loading of a resource at self ("script-src moz-extension://fb4eb661-52e1-430e-aee1-b91f0297750b"). Source: call to eval() or related function blocked by CSP.  background.js:20
Unchecked lastError value: Error: No matching message handler  ExtensionCommon.jsm:409
Error: Could not establish connection. Receiving end does not exist.  undefined
"Download the React DevTools for a better development experience: https://fb.me/react-devtools"  popup.js:59505:9
"Warning: Accessing PropTypes via the main React package is deprecated, and will be removed in  React v16.0. Use the latest available v15.* prop-types package from npm instead. For info on usage, compatibility, migration and more, see https://fb.me/prop-types-docs"  popup.js:76948:7
"Download the React DevTools for a better development experience: https://fb.me/react-devtools"  popup.js:59505:9
"Warning: Accessing PropTypes via the main React package is deprecated, and will be removed in  React v16.0. Use the latest available v15.* prop-types package from npm instead. For info on usage, compatibility, migration and more, see https://fb.me/prop-types-docs"  popup.js:76948:7
"Download the React DevTools for a better development experience: https://fb.me/react-devtools"  popup.js:59505:9
"Warning: Accessing PropTypes via the main React package is deprecated, and will be removed in  React v16.0. Use the latest available v15.* prop-types package from npm instead. For info on usage, compatibility, migration and more, see https://fb.me/prop-types-docs"  popup.js:76948:7
"Download the React DevTools for a better development experience: https://fb.me/react-devtools"  popup.js:59505:9
"Warning: Accessing PropTypes via the main React package is deprecated, and will be removed in  React v16.0. Use the latest available v15.* prop-types package from npm instead. For info on usage, compatibility, migration and more, see https://fb.me/prop-types-docs"  popup.js:76948:7
"Download the React DevTools for a better development experience: https://fb.me/react-devtools"  popup.js:59505:9
"Warning: Accessing PropTypes via the main React package is deprecated, and will be removed in  React v16.0. Use the latest available v15.* prop-types package from npm instead. For info on usage, compatibility, migration and more, see https://fb.me/prop-types-docs"  popup.js:76948:7
SETTING SYMBOL AND DECIMALS Object { symbol: Object[1], decimals: Object[1] }  popup.js:94974:15
"Download the React DevTools for a better development experience: https://fb.me/react-devtools"  popup.js:59505:9
"Warning: Accessing PropTypes via the main React package is deprecated, and will be removed in  React v16.0. Use the latest available v15.* prop-types package from npm instead. For info on usage, compatibility, migration and more, see https://fb.me/prop-types-docs"  popup.js:76948:7
"Download the React DevTools for a better development experience: https://fb.me/react-devtools"  popup.js:59505:9
"Warning: Accessing PropTypes via the main React package is deprecated, and will be removed in  React v16.0. Use the latest available v15.* prop-types package from npm instead. For info on usage, compatibility, migration and more, see https://fb.me/prop-types-docs"  popup.js:76948:7
SETTING SYMBOL AND DECIMALS Object { symbol: Object[1], decimals: Object[1] }  popup.js:94974:15
(node) warning: possible EventEmitter memory leak detected. 11 listeners added. Use emitter.setMaxListeners() to increase limit.  background.js:57972:7
	[388]</EventEmitter.prototype.addListener moz-extension://de7e5cfc-0466-46b7-9096-a3a4b6a07a9a/scripts/background.js:57972:7
	subscribe moz-extension://de7e5cfc-0466-46b7-9096-a3a4b6a07a9a/scripts/background.js:73312:7
	ObsStoreStream moz-extension://de7e5cfc-0466-46b7-9096-a3a4b6a07a9a/scripts/background.js:73401:5
	asStream moz-extension://de7e5cfc-0466-46b7-9096-a3a4b6a07a9a/scripts/background.js:73378:10
	setupPublicConfig moz-extension://de7e5cfc-0466-46b7-9096-a3a4b6a07a9a/scripts/background.js:5813:12
	setupUntrustedCommunication moz-extension://de7e5cfc-0466-46b7-9096-a3a4b6a07a9a/scripts/background.js:5751:7
	connectRemote moz-extension://de7e5cfc-0466-46b7-9096-a3a4b6a07a9a/scripts/background.js:301:7
console.trace():  background.js:57978
	[388]</EventEmitter.prototype.addListener moz-extension://de7e5cfc-0466-46b7-9096-a3a4b6a07a9a/scripts/background.js:57978:9
	subscribe moz-extension://de7e5cfc-0466-46b7-9096-a3a4b6a07a9a/scripts/background.js:73312:7
	ObsStoreStream moz-extension://de7e5cfc-0466-46b7-9096-a3a4b6a07a9a/scripts/background.js:73401:5
	asStream moz-extension://de7e5cfc-0466-46b7-9096-a3a4b6a07a9a/scripts/background.js:73378:10
	setupPublicConfig moz-extension://de7e5cfc-0466-46b7-9096-a3a4b6a07a9a/scripts/background.js:5813:12
	setupUntrustedCommunication moz-extension://de7e5cfc-0466-46b7-9096-a3a4b6a07a9a/scripts/background.js:5751:7
	connectRemote moz-extension://de7e5cfc-0466-46b7-9096-a3a4b6a07a9a/scripts/background.js:301:7
Error: Could not establish connection. Receiving end does not exist.  undefined
`

**Comment by tmashuang** [comment]
Related to #2748. Anyone having an issue with rendering tokens, the potential fix would be to refresh the token list. Switching to the 'Send' tab and back to the 'Token' tab will refresh the list.

**Comment by Christophe80** [comment]
Tried switching, same problem...

**Comment by tmashuang** [comment]
Also, token balances can be viewed when a MetaMask account is synced with MyEtherWallet.

**Comment by bgadrian** [comment]
I can confirm the bug on Firefox 59, but it WORKS on Chrome 62 (I can add a new token on chrome).

**Comment by Christophe80** [comment]
In etherwallet and etherscan I can see tokens, but Metamask doesn't recognise them... [https://www.dropbox.com/s/32ndx4oqnxdpm55/Naamloos.mov?dl=0](url) I'm using Chrome...

**Comment by danfinlay** [comment]
Our token view is fairly brittle right now, we will be making it more robust in #2748, so closing this issue as a duplicate of that one.

**Comment by kennblvnp** [comment]
there is a reason why this is happening, its because the token is not migrated to the same ip address and port in **truffle.js**

At least its what I have observed.

**Comment by Kruln** [comment]
> There is a reason why this is happening, its because the token is not migrated to the same IP Address and Port. Check **truffle.js**
> 
> At least its what I have observed.

Yes, here i found my answer. I switched off mij vpn and was able to add token right away.
